onnxruntime/core/providers/webnn/builders/helper.cc

```diff
@@ -280,8 +283,9 @@ bool IsDataTypeSupportedByWebNNOp(const std::string_view onnx_op_type,
                           << webnn_input_output_name << "]";
     return false;
   }
-  if (!IsSupportedDataType(
-          onnx_data_type, wnn_limits[std::string(webnn_op_type)][std::string(webnn_input_output_name)]["dataTypes"])) {
+  if (!IsSupportedDataType(onnx_data_type,
+                           wnn_limits[std::string(webnn_op_type)][std::string(webnn_input_output_name)]["dataTypes"],
+                           wnn_limits)) {
```

Comment:
It feels odd to pass a subfield of `wnn_limits` but then also to pass the entire `wnn_limits` in the very next parameter, because then why not just pass the full `wnn_limits` along with the `webnn_op_type` and `webnn_input_output_name`?

Reply:
Good point, while in following usage, the `constant` doesn't have an input/output name, or shall we just pass empty string as its input/output name and skip it if it is empty?

```js
if (IsSupportedDataType(data_type, wnn_limits_["constant"]["dataTypes"], wnn_limits_)) {
```